code/mob/living/critter/small_animal.dm

```diff
@@ -1855,6 +1855,150 @@ var/list/mob_bird_species = list("smallowl" = /mob/living/critter/small_animal/b
 		qdel(friends)
 		return ..()
 
+/* =================================================== */
+/* ------------------- Rattlesnake ------------------- */
+/* =================================================== */
+
+/mob/living/critter/small_animal/rattlesnake
+	name = "rattlesnake"
+	real_name = "rattlesnake"
+	blood_id = "blood"
+	desc = "A snake. With a rattle. A rattlesnake."
+	icon_state = "rattlesnake"
+	icon_state_dead = "rattlesnake_dead"
+	speechverb_say = "hisses"
+	speechverb_exclaim = "rattles"
+	speechverb_ask = "hisses"
+	health_brute = 20
+	health_burn = 20
+	density = 0
+	flags = TABLEPASS
+	fits_under_table = 1
+	can_lie = 0
+	ai_type = /datum/aiHolder/rattlesnake
+	is_npc = TRUE
+	var/aggressive = TRUE
+	var/list/friends = list()
+	add_abilities = list(/datum/targetable/critter/wasp_sting/snake_bite)
+
+	New()
+		..()
+		src.event_handler_flags |= USE_PROXIMITY
+
+	setup_hands()
+		..()
+		var/datum/handHolder/HH = hands[1]
+		HH.limb = new /datum/limb/mouth
+		HH.icon = 'icons/mob/critter_ui.dmi'
+		HH.icon_state = "mouth"
+		HH.name = "mouth"
+		HH.limb_name = "teeth"
+		HH.can_hold_items = 0
+
+	attackby(obj/item/I, mob/M)
+		if(istype(I, /obj/item/reagent_containers/food/snacks) && ishuman(M) && !isdead(src))
+			src.visible_message("[M] feeds \the [src] some [I].", "[M] feeds you some [I].")
+			for(var/damage_type in src.healthlist)
+				var/datum/healthHolder/hh = src.healthlist[damage_type]
+				hh.HealDamage(5)
+				src.health_brute = min(60, src.health_brute + 6)
+				src.health_burn = min(60, src.health_burn + 6)
+			if(M in src.friends)
+				src.emote("rattle")
+			else
+				if(prob(20))
+					friends += M
+					src.visible_message("[src] hisses happily at the \the [I], and seems a little friendlier with [M].")
+				else
+					src.visible_message("<span class='notice'>[src] hated \the [I] and bit [M]'s hand!</span>")
+					random_brute_damage(M, rand(6,12),1)
+					src.emote("hiss")
+					M.emote("scream")
+			I.Eat(src, src, TRUE)
+			return
+		. = ..()
+
+	attack_hand(mob/M)
+		if ((M.a_intent != INTENT_HARM) && (M in src.friends))
+			if(M.a_intent == INTENT_HELP && src.aggressive)
+				src.visible_message("<span class='notice'>[M] pats [src] on the head in a soothing way. It won't attack anyone now.</span>")
+				src.aggressive = FALSE
+				return
+			else if((M.a_intent == INTENT_DISARM) && !src.aggressive)
+				src.visible_message("<span class='notice'>[M] shakes [src] to awaken it's killer instincts!</span>")
+				src.aggressive = TRUE
+				return
+		..()
+
+	specific_emotes(var/act, var/param = null, var/voluntary = 0)
+		switch (act)
+			if ("scream","hiss")
+				if (src.emote_check(voluntary, 50))
+					playsound(src.loc, 'sound/voice/animal/cat_hiss.ogg', 50, 1, channel=VOLUME_CHANNEL_EMOTE)
+					return "<span class='emote'><b>[src]</b> hisses!</span>"
+			if ("rattle", "snap")
+				if (src.emote_check(voluntary, 50))
```

Comment:
```suggestion
				if (src.emote_check(voluntary, 50) && icon_state == "rattlesnake")
```
just to make sure you're not currently rattling already, so the `SPAWN` doesn't get messed up